ash/src/device.rs

```diff
@@ -2107,21 +2101,16 @@ impl Device {
         &self,
         image: vk::Image,
     ) -> Vec<vk::SparseImageMemoryRequirements> {
-        let mut count = 0;
-        self.device_fn_1_0.get_image_sparse_memory_requirements(
-            self.handle(),
-            image,
-            &mut count,
-            ptr::null_mut(),
-        );
-        let mut data = Vec::with_capacity(count as usize);
-        self.device_fn_1_0.get_image_sparse_memory_requirements(
-            self.handle(),
-            image,
-            &mut count,
-            data.as_mut_ptr(),
-        );
-        data.set_len(count as usize);
-        data
+        read_into_vector(|count, data| {
+            self.device_fn_1_0.get_image_sparse_memory_requirements(
+                self.handle(),
+                image,
+                count,
+                data,
+            );
+            vk::Result::SUCCESS
+        })
+        // The closure always returns SUCCESS
+        .unwrap()
```

Comment:
This helper function is not ideal for Vulkan functions without error codes, but it's does save quite a few lines.

Reply:
This seems fine to me. It should optimize out trivially.

Reply:
Just seems ugly to me - but the original duplicated call more so and is more error-prone at the same time.

Will keep it this way, shall I duplicate this comment to every (4 or so, iirc) occurrence of this pattern or leave it out entirely?

Reply:
No strong preference; I think it's obvious enough from context that the comment isn't absolutely necessary, but it saves a double-take.